packages/data-flow/src/interfaces/eventsRegistry.interface.ts

```diff
@@ -0,0 +1,18 @@
+import { AnyEvent, ContractName, ProcessorEvent } from "@grants-stack-indexer/shared";
+
+/**
+ * The events registry saves as a checkpoint to the last processed event by the system.
```

Comment:
the registry itself is the checkpoint, you query the registry for getting the last processed event (however at current MVP stage, since it's in memory on shutdown you lose tracking of course)